src/test/regress/sql/explain_analyze.sql

```diff
@@ -7,6 +7,16 @@
 -- s/Buckets: \d+/Buckets: ###/
 -- m/Batches: \d+/
 -- s/Batches: \d+/Batches: ###/
+-- m/Planning Time: [0-9.]+ ms/
```

Comment:
Max executor memory usage can be not only on the `(seg0)`. And it is not covered by the matchsubs and can lead to flaky results. And I was able to reproduce it on my side:
```diff
diff -I HINT: -I CONTEXT: -I GP_IGNORE: -U3 /usr/re/share/review/ws44-ADBDEV-6393/gpdb/src/test/regress/expected/explain_analyze.out /usr/re/share/review/ws44-ADBDEV-6393/gpdb/src/test/regress/results/explain_analyze.out
--- /usr/re/share/review/ws44-ADBDEV-6393/gpdb/src/test/regress/expected/explain_analyze.out	2024-10-11 09:41:14.209145891 +1000
+++ /usr/re/share/review/ws44-ADBDEV-6393/gpdb/src/test/regress/results/explain_analyze.out	2024-10-11 09:41:14.213145897 +1000
@@ -113,7 +113,7 @@
  Planning Time: #.### ms
    (slice0)    Executor memory: ### bytes.
    (slice1)    Executor memory: ### bytes avg x 3 workers, ### bytes max (seg0).  Work_mem: ### bytes max.
-   (slice2)    Executor memory: ### bytes avg x 3 workers, ### bytes max (seg0).
+   (slice2)    Executor memory: ### bytes avg x 3 workers, ### bytes max (seg1).
  Memory used:  ###B
  Execution Time: #.### ms
 GP_IGNORE:(17 rows)
```

Reply:
> And I was able to reproduce it on my side

Same on CI

Reply:
Fixed